### ISSUE TITLE
Proposal to treat each run through a scenario outline as a distinct scenario.

### ISSUE THREAD

**Issue by wholewheattoast** [npilon/planterbox#11]
Currently when you iterate through a scenario outline like:

```
Scenario Outline: Winslow pages requiring user to be signed out
        Given I go to "<foo>"
       Then there should be "<bar>"

        Examples:
            | foo                  |  bar | 
            | example.com | Hi |
            | google.com   | how are you |
```

The hooks ie `@hook('before', 'scenario')` only fire at the beginning/end of the entire Scenario Outline.

In my mind it seems that each 'run' through a scenario outline should be treated as a distinct scenario.  Behavior that should be triggered before or after a scenario (as defined by the hooks) should be triggered for each outline example row.

The gherkin syntax isn't 100% clear on this https://github.com/cucumber/cucumber/wiki/Scenario-Outlines but the example at the top of the page that starts with "Copying and pasting scenarios to use different values quickly becomes tedious and repetitive:" leads some credibility to my assumptions.

Let me know if you need more details.

**Comment by npilon** [comment]
Should be addressed in 0.6

**Comment by wholewheattoast** [comment]
tests pass yay.
